sonar-plugin-api/src/main/java/org/sonar/api/server/rule/GenericConceptKeyFormat.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Sonar Plugin API
+ * Copyright (C) 2009-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.rule;
+
+/**
+ * The characters allowed for the generic concepts keys are (lowercase) alphanumerics with dash (-), underscore (_) and dot (.)
```

Comment:
one more place to update :P I wonder if you can't simply make a reference to the constant here, this API is for developers after all and this would prevent such issues...

Reply:
I have updated the javadoc to include automatically the regex pattern